# live Code Phase 0

## 1. Introduction

Name : Amabel Nabila <br>
Batch : 012

## 2. Import Libraries

In [1]:
import pandas as pd
import numpy as np

## 3. Data Loading

In [17]:
df = pd.read_excel (r'NMC1-9.xlsx')

In [18]:
pd.set_option("display.max_columns", None)
df.head()

time(s)    V_m  Current (A)  Cycle       Cap
0        0  4.098        0.997      1  2.840817
1        1  4.096        0.998      1  2.840817
2        2  4.094        0.999      1  2.840817
3        3  4.093        0.998      1  2.840817
4        4  4.092        0.999      1  2.840817

Keterangan kolom:

time(s): Detik ke- pengukuran <br>
V_m: Voltage/tegangan yang terukur tiap waktu <br>
Current (A): Kuat arus terukur tiap waktu <br>
Cycle: Nomor siklus charge-discharge baterai <br>
Cap: Kapasitas baterai tiap waktu 


## 4. Questions and Answers

**a. Buatlah dua fungsi eksplisit masing-masing memuat perhitungan hambatan (R) dan daya (P) yang nantinya akan digunakan untuk perhitungan**

In [19]:
df.rename(columns={'V_m': 'V', 'Current (A)' :'I'}, inplace=True)

In [20]:
df.head()

time(s)      V      I  Cycle       Cap
0        0  4.098  0.997      1  2.840817
1        1  4.096  0.998      1  2.840817
2        2  4.094  0.999      1  2.840817
3        3  4.093  0.998      1  2.840817
4        4  4.092  0.999      1  2.840817

In [46]:
df['V'].sum()

328232.673

In [47]:
df['I'].sum()

90915.061

In [48]:
def hambatan(V,I):
    print(V/I)
hambatan(328232.673, 90915.061)

3.610322309523611


In [49]:
def daya(V,I):
    return(V*I)
daya (328232.673, 90915.061)

29841293487.988056

After the total sum of each voltage (V) and current (I), the results are entered into the formula for resistance and daya/power every second. Based on the results, the result of the calculation of the resistance is 3.610322309523611 and the result of the calculation of the power is 29841293487.988056

**b. Buat kolom baru dengan nama kolom R/P berisikan hasil perhitungan daya/hambatan dengan kriteria untuk Kapasitas >=2.5 hitung daya, Kapasitas <2.5 hitung nilai hambatan. (Hint: Gunakan for loop dan if statement untuk mengerjakan perintah nomor b, kamu bisa menggunakan list untuk menyimpan perhitungan sementara, setelahnya dapat dimasukkan ke kolom R/P).**

R/P = daya/hambatan

In [54]:
 3.610322309523611/29841293487.988056

1.2098410918336585e-10

In [61]:
df.head(2)

time(s)      V      I  Cycle       Cap
0        0  4.098  0.997      1  2.840817
1        1  4.096  0.998      1  2.840817

In [63]:
hasil = []
for idx in range(0,len(df)):
    if Cap >=2.5:
        daya = df.iloc[idx, 1] * df.iloc[idx, 2]
    elif Cap <2.5:
        hambatan =  df.iloc[idx, 1] / df.iloc[idx, 2]
        hasil.append(Cap)

NameError: name 'Cap' is not defined

c. Hitunglah rata-rata nilai kapasitas untuk arus = 1 Ampere dan cycle ganjil.

In [29]:
df[
    (df['I'] == 1) &
    (df['Cycle'] % 2 == 0) 
].mean()

time(s)    45977.148759
V              3.609307
I              1.000000
Cycle          5.024825
Cap            2.806838
dtype: float64

Based on the above results, it can be seen that the average value for the current equal to 1 ampere (which is initialized in the table with **I**) is 1.0. Moreover, the average value for odd cycle is 5.02.